cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -134,32 +134,40 @@ func RemoveStackedEtcdMemberFromCluster(client clientset.Interface, cfg *kubeadm
 // CreateStackedEtcdStaticPodManifestFile will write local etcd static pod manifest file
 // for an additional etcd member that is joining an existing local/stacked etcd cluster.
 // Other members of the etcd cluster will be notified of the joining node in beforehand as well.
-func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifestDir, patchesDir string, nodeName string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, isDryRun bool) error {
+func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifestDir, patchesDir string, nodeName string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, isDryRun bool, certificatesDir string) error {
 	// creates an etcd client that connects to all the local/stacked etcd members
 	klog.V(1).Info("creating etcd client that connects to etcd pods")
-	etcdClient, err := etcdutil.NewFromCluster(client, cfg.CertificatesDir)
+	etcdClient, err := etcdutil.NewFromCluster(client, certificatesDir)
 	if err != nil {
 		return err
 	}
 
 	etcdPeerAddress := etcdutil.GetPeerURL(endpoint)
 
-	klog.V(1).Infof("[etcd] Adding etcd member: %s", etcdPeerAddress)
 	var cluster []etcdutil.Member
-	cluster, err = etcdClient.AddMember(nodeName, etcdPeerAddress)
-	if err != nil {
-		return err
+	if isDryRun {
+		fmt.Printf("[dryrun] Would add etcd member: %s\n", etcdPeerAddress)
+	} else {
+		klog.V(1).Infof("[etcd] Adding etcd member: %s", etcdPeerAddress)
+		cluster, err = etcdClient.AddMember(nodeName, etcdPeerAddress)
+		if err != nil {
+			return err
+		}
+		fmt.Println("[etcd] Announced new etcd member joining to the existing etcd cluster")
+		klog.V(1).Infof("Updated etcd member list: %v", cluster)
 	}
 
-	fmt.Println("[etcd] Announced new etcd member joining to the existing etcd cluster")
-	klog.V(1).Infof("Updated etcd member list: %v", cluster)
-
 	fmt.Printf("[etcd] Creating static Pod manifest for %q\n", kubeadmconstants.Etcd)
 
 	if err := prepareAndWriteEtcdStaticPod(manifestDir, patchesDir, cfg, endpoint, nodeName, cluster, isDryRun); err != nil {
```

Comment:
passing a nil cluster seems fine here, due to:
https://github.com/kubernetes/kubernetes/blob/94769757079c71ef2377260b84c0c81ec2e671d8/cmd/kubeadm/app/phases/etcd/local.go#L233-L234

https://github.com/kubernetes/kubernetes/blob/94769757079c71ef2377260b84c0c81ec2e671d8/cmd/kubeadm/app/phases/etcd/local.go#L173

https://github.com/kubernetes/kubernetes/blob/94769757079c71ef2377260b84c0c81ec2e671d8/cmd/kubeadm/app/phases/etcd/local.go#L251-L253